packages/input-selection/test/util/index.ts

```diff
@@ -7,3 +8,13 @@ export * as SelectionConstraints from './selectionConstraints';
 export const asAssetId = (x: string): Cardano.AssetId => x as unknown as Cardano.AssetId;
 export const asPaymentAddress = (x: string): Cardano.PaymentAddress => x as unknown as Cardano.PaymentAddress;
 export const asTokenMap = (elements: Iterable<[Cardano.AssetId, bigint]>) => new Map<Cardano.AssetId, bigint>(elements);
+
+export class MockChangeAddressResolver implements ChangeAddressResolver {
+  async resolve(selection: Selection) {
+    return selection.change.map((txOut) => {
+      txOut.address =
```

Comment:
This also updates the input param `selection.change` as a side effect. Is this the intended behavior?